### PR TITLE
added translation for OIDC choices for Client authentication

### LINKED ISSUE #3080: OIDC Identity provider has ugly choices for `Client authentication`
### Describe the bug

Dropdown choices are using internal values like, `clientAuthentications.client_secret_basic`.

The dropdown should show friendly choices like in the old console.  For instance, `Client secret sent as basic auth`.

### Version

20.0.0-SNAPSHOT

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ikhomyn**
## Motivation
Fixes #3080 

## Checklist:

- [X] Code has been tested locally by PR requester
- [ ] User-visible strings are using the react-i18next framework (useTranslation)
- [ ] Help has been implemented
- [ ] axe report has been run and resulting a11y issues have been resolved
- [ ] Unit tests have been created/updated

### CODE REVIEW

File: src/identity-providers/add/OIDCAuthentication.tsx [hunk 1]
```diff
@@ -11,19 +11,18 @@ import {
...
-  "private_key_jwt",
-];
-
 export const OIDCAuthentication = ({ create = true }: { create?: boolean }) => {
   const { t } = useTranslation("identity-providers");
 
   const { control } = useFormContext();
   const [openClientAuth, setOpenClientAuth] = useState(false);
 
+  const clientAuthentications = [
```
Comment (jonkoops) [inline src/identity-providers/add/OIDCAuthentication.tsx:20]: Why move these into the component if they can be defined as a constant outside of it?

File: src/identity-providers/add/OIDCAuthentication.tsx [hunk 2]
```diff
@@ -42,24 +41,24 @@ export const OIDCAuthentication = ({ create = true }: { create?: boolean }) => {
...
-              toggleId="clientAuthMethod"
+              toggleId="clientAuthentication"
               required
               onToggle={() => setOpenClientAuth(!openClientAuth)}
               onSelect={(_, value) => {
                 onChange(value as string);
                 setOpenClientAuth(false);
               }}
-              selections={value}
+              selections={t(`${value}`)}
```
Comment (jonkoops) [inline src/identity-providers/add/OIDCAuthentication.tsx:56]: Why would the selections need to be translated?

Comment (ikhomyn) [reply src/identity-providers/add/OIDCAuthentication.tsx]: Hi @jonkoops , yes, you are right, there is not sense. I've updated both suggestions. Thanks!

File: src/identity-providers/add/OIDCAuthentication.tsx [hunk 3]
```diff
@@ -11,11 +11,11 @@ import {
...
-const clientAuthenticationTypes = [
-  "client_secret_post",
-  "client_secret_basic",
-  "client_secret_jwt",
-  "private_key_jwt",
+const clientAuthentications = [
+  "clientAuth_post",
+  "clientAuth_basic",
+  "clientAuth_secret_jwt",
+  "clientAuth_privatekey_jwt",
```
Comment (edewit) [inline src/identity-providers/add/OIDCAuthentication.tsx:18]: Can't change these values as they saved and the backend expects them to be like this

Comment (edewit) [reply src/identity-providers/add/OIDCAuthentication.tsx:18]: It would be better to change the translation keys instead

**Review by edewit — CHANGES_REQUESTED** [review]

File: src/identity-providers/add/OIDCAuthentication.tsx (hunk 3 above)
Comment (ikhomyn) [reply src/identity-providers/add/OIDCAuthentication.tsx]: I see. I've changed them back and updated translation keys. Thank you @edewit

**Review by edewit — APPROVED** [review]

**Review by jonkoops — APPROVED** [review]

**Review by edewit — APPROVED** [review]

**Pull request merged**